plugins/antispam/antispam_plugin.cpp

```diff
@@ -58,35 +90,42 @@ PCL void OnMessageSent(char *message,int slot, qboolean *show, int type){
 		data.players[slot].lastMessage = t;
 	    return;
 	}
+
+
 	// Max messages per minute check
 	int i,j=0;
-	for(i=0;( i<ANTISPAM_MAXMESSAGES) && (data.players[slot].messages[i] != 0) && (t - data.players[slot].messages[i] > 60);++i);
+	for (i=0;( i<ANTISPAM_MAXMESSAGES) && (data.players[slot].messages[i] != 0) && (t - data.players[slot].messages[i] > 60);++i);
 	// the loop above iterates on the messages array until it's end, value of 0 or message older than 1 minute
 	// now the i value holds the number of messages in the messages array which are older than 60 seconds + 1
 	// we need to remove those
-	if(i!=0){
-	    for(j=0;j<ANTISPAM_MAXMESSAGES && i<ANTISPAM_MAXMESSAGES && data.players[slot].messages[j] != 0;++i,++j);{
+	if(i!=0)
+	{
+	    for(j=0;j<ANTISPAM_MAXMESSAGES && (i < (ANTISPAM_MAXMESSAGES)) && data.players[slot].messages[j] != 0;++i,++j);
+		{
 		data.players[slot].messages[j] = data.players[slot].messages[i];
 		data.players[slot].messages[i] = 0;
 	    }
 	}
-	Plugin_ChatPrintf(slot,"Debug: %d, %d, %d, %d.\n",j,i,data.maxMPM->integer,data.maxMPM->latchedinteger);
 	for(i=j;i<ANTISPAM_MAXMESSAGES;++i){
 	    data.players[slot].messages[i] = 0;
 	}
 
 	// j now holds the value of how many messages this player has sent in the last 60 seconds. If it is too much - dont show the msg
 	// if it is ok - show the message and add time to messages[j]
-	if(j<data.maxMPM->integer){
+	if(j<ANTISPAM_MAXMESSAGES)
```

Comment:
what's the point of replacing maxMPM with ANTISPAM_MAXMESSAGES? you just make the number of message unconfigurable that way. you even documented maxMPM in your readme, but now it is useless.